client/core/core.go

```diff
@@ -539,21 +539,35 @@ func (c *Core) CreateWallet(appPW, walletPW []byte, form *WalletForm) error {
 		return fmt.Errorf("wallet password encryption error: %v", err)
 	}
 
+	walletInfo, err := asset.Info(assetID)
+	if err != nil {
+		return err
+	}
+
 	var settings map[string]string
 	if form.ConfigText == "" {
-		var defaultWalletCfgPath string
-		defaultWalletCfgPath, err = asset.DefaultConfigPath(assetID)
-		if err != nil {
-			return fmt.Errorf("cannot use default wallet config path: %v", err)
-		}
-		settings, err = config.Parse(defaultWalletCfgPath)
+		settings, err = config.Parse(walletInfo.DefaultConfigPath)
 	} else {
 		settings, err = config.Parse([]byte(form.ConfigText))
 	}
 	if err != nil {
 		return fmt.Errorf("error parsing config: %v", err)
 	}
 
+	// Remove unused key-values from parsed settings before saving to db.
+	// Especially necessary if settings was parsed from a config file, b/c
+	// config files usually define more key-values than we need.
+	expectedKeys := make(map[string]bool, len(walletInfo.ConfigOpts))
+	for _, option := range walletInfo.ConfigOpts {
+		expectedKeys[option.Key] = true
+	}
+	for key := range settings {
+		expected := expectedKeys[key]
+		if !expected {
+			delete(settings, key)
```

Comment:
Actually, do we need to think about text case here?  Is the downstream settings processing case insensitive?  Or is this already handled by config.Parse?

Reply:
`expectedKeys` should probably use lowercased keys, and lowercased `settings` keys also used to check. I think while `config.Parse` may treat keys as case-insensitive, the keys returned in the `settings` map may not be case-insensitive. Will check thoroughly.

Reply:
TL;DR: downstream settings processing is **case sensitive**. Every asset's `walletInfo.ConfigOpts` should list keys in the same case as the ini tag of the asset's config struct AND user-provided config / the default cfg file must define values using keys in the same case as the config struct.

Epistle:
I believe by downstream settings processing, you mean the different asset-specific functions that read config values from the `settings` map to connect to the wallet.

Those functions call `config.Unmapify` to read the settings map into a config object. Keys are case sensitive currently, so a map entry `Key=value` will not be "unmapified" into an object with property `Key string 'ini="key"'`.

For that reason, it makes sense to delete wrongly-cased keys from the settings map before the map is processed downstream and subsequently saved to db. This delete code currently does that.

The caveat is that each asset's `walletInfo.ConfigOpts` **MUST** list option keys in the same case as defined in the config struct. That way, we know that any keys we delete from the settings map here is really useless to us because the settings processing code downstream will ignore it.

This caveat is inherently taken care of by the `config.Options` function. But should an asset not use that function to define its `walletInfo.ConfigOpts`, we could easily run into trouble, which is why it should be necessary to add tests for such an asset to ensure that their manually defined `walletInfo.ConfigOpts` is in sync with their config struct.

Reply:
Alternatively, we can go the case insensitive route and allow user-provided settings / config files to define keys in any case.

The caveat is that keys would be converted to lowercase during parsing, which actually shouldn't be a problem as configs can still define keys in any case. (It just makes checking parsed configs in tests a little messy as the original data may define mixed case, but the parsed config would have lowercase).

I've implemented this anyways, will push so we can peruse.

Reply:
I like the case insensitive option as case mismatch errors can be frustrating to identify for the user.
But I think the question is are the wallet's case insensitive or would there be a case where the wallet might actually require `someSetting=x`?